src/FSharp.Data.GraphQL.Server.Middlewares/TypeSystemExtensions.fs

```diff
@@ -1,7 +1,68 @@
 namespace FSharp.Data.GraphQL.Server.Middlewares
 
 open FSharp.Data.GraphQL.Types
+open FSharp.Data.GraphQL.Ast
 
+/// A function that checks if a directive should be used in the exection of a query, or changed to a new directive.
+type DirectiveChooser = Directive -> Directive option
```

Comment:
This function will be used to choose Directives. We can opt to transform them into a new Directive, or remove them from the operation. For example, if we remove a `defer` directive by returning `None`, all deferred fields will be returned directly.